packages/components/src/lib/utility.ts

```diff
@@ -0,0 +1,33 @@
+import { result } from "@quri/squiggle-lang";
+
+export function flattenResult<a, b>(x: result<a, b>[]): result<a[], b> {
```

Comment:
I like the idea of us pulling out more functional functions into a lib like this. I feel like there might be some other standard lib for this though (but that could wait for later)